rockcraft/extensions/go.py

```diff
@@ -0,0 +1,213 @@
+
+        install_app_part = {
+            "plugin": "go",
+            "source": ".",
+            "organize": organize,
+        }
+
+        if not self._check_go_overriden():
+            build_snaps = install_app.get("build-snaps", [])
+            build_snaps.append("go")
+            install_app_part["build-snaps"] = build_snaps
+
+        install_app_part["stage"] = list(organize.values())
+        if build_environment:
+            install_app_part["build-environment"] = build_environment
+
+        return install_app_part
+
+    def _check_go_overriden(self):
+        """Check if the user overrode the go snap or package for the build step."""
+        build_snaps = self._get_nested(
+            self.yaml_data, "parts.go-framework/install-app.build-snaps"
+        )
+        if build_snaps:
+            for snap in build_snaps:
+                if snap.startswith("go"):
+                    return True
+        build_packages = self._get_nested(
+            self.yaml_data, "parts.go-framework/install-app.build-packages"
+        )
+        if build_packages:
+            for package in build_packages:
+                if package in ["gccgo-go", "golang-go"]:
+                    return True
+        return False
+
+    def _gen_install_assets_part(self):
+        """Generate assets-stage part for extra assets in the project."""
+        # if stage is not in exclude mode, use it to generate organize
+        if (
+            self._assets_stage
+            and self._assets_stage[0]
+            and self._assets_stage[0][0] != "-"
+        ):
+            renaming_map = {
+                os.path.relpath(file, "app"): file for file in self._assets_stage
+            }
+        else:
+            return None
+
+        return {
+            "plugin": "dump",
+            "source": ".",
+            "organize": renaming_map,
+            "stage": self._assets_stage,
+        }
+
+    @property
+    def _assets_stage(self):
+        """Return the assets stage list for the Go project."""
+        user_stage = self._get_nested(self.yaml_data, "parts.go-framework/assets").get(
+            "stage", []
+        )
+
+        if not all(re.match("-? *app/", p) for p in user_stage):
+            raise ExtensionError(
+                "go-framework extension requires the 'stage' entry in the "
+                "go-framework/assets part to start with app",
+                docs_url="https://documentation.ubuntu.com/rockcraft/en/stable/reference/extensions/go-framework",
+                logpath_report=False,
+            )
+        if not user_stage:
+            user_stage = [
+                f"app/{f}"
+                for f in (
+                    "migrate",
+                    "migrate.sh",
+                    "static",
+                    "templates",
+                )
+                if (self.project_root / f).exists()
+            ]
+        return user_stage
+
+    def _get_nested(self, obj: dict, path: str) -> dict:
```

Comment:
two comments here: 
1) I believe dots are valid in part names, so this is fragile. How about changing it to a list of names instead? E.g. the signature could be `def _get_nested(self, obj, *parts: str)`
2) This says it returns `dict` but `build-snaps` is a list. Seems like this should return `Any`

Reply:
Updated to be a list instead of a string so dots could be used in part names.

I changed to not get `build-snaps` directly, as it can be a list, and instead get the object. Otherwise I would need to complicate the _get_nested function to check for the type (and as it is a very simple helper, I preferred not to handle all cases unless more needs arise).